resources/js/components/ui/Modal/Modal.tsx

```diff
@@ -0,0 +1,53 @@
+import * as dialog from "@zag-js/dialog";
+import {normalizeProps, useMachine} from "@zag-js/solid";
+import {ParentComponent, Show, createComputed, createMemo, createUniqueId} from "solid-js";
+import {Portal} from "solid-js/web";
+import s from "./Modal.module.scss";
+
+interface Props {
+  title?: string;
+  open: boolean;
+  /**
+   * Event triggered when the Esc key is pressed or when the user clicks outside of the modal.
+   * If specified, it can react to that event by closing the dialog (i.e. setting the open prop
+   * to false).
+   */
+  onEscape?: (reason: "key" | "click") => void;
+}
+
+export const Modal: ParentComponent<Props> = (props) => {
+  const [state, send] = useMachine(
+    dialog.machine({
+      closeOnOutsideClick: false,
+      closeOnEsc: false,
+      onEsc: () => props.onEscape?.("key"),
+      onOutsideClick: () => props.onEscape?.("click"),
+      id: createUniqueId(),
+    }),
+  );
+  const api = createMemo(() => dialog.connect(state, send, normalizeProps));
+  createComputed(() => {
```

Comment:
czemu createComputed, a nie createRenderEffect? createComputed jest bardziej "niskopoziomowe" dla solida, a oba dają w tym przypadku praktycznie to samo

Reply:
No tak, createComputed oczywiście nie ma sensu, odruchowo tak napisałem. Natomiast nie ogarniam trochę praktycznej różnicy między createEffect a createRenderEffect więc użyję tego co mówisz.

Reply:
https://github.com/solidjs/solid-workgroup/discussions/2
co ciekawe, tu autor solida pisze, że planuje createComputed uczynić deprecated ale nie rozumiem z jakiego powodu, więc createRenderEffect to to co chcemy używać